CI timeout during package upgrade in xenial staging scenario

## Description

When upgrading distribution packages in staging, the CI run times out due to lack of output. This is likely due to a large amount of packages on the both app and mon due to nested virt

## Steps to Reproduce

Observe CI output of these runs

- https://circleci.com/gh/freedomofpress/securedrop/23846
- https://circleci.com/gh/freedomofpress/securedrop/23827
- https://circleci.com/gh/freedomofpress/securedrop/23886

```
    TASK [common : Perform safe upgrade to ensure all the packages are updated.] ***
make: *** [ci-go-xenial] Terminated
Warning: Permanently added '35.230.112.0' (ECDSA) to the list of known hosts.

scp: /home/sdci/securedrop-source/junit/*xml: No such file or directory
Too long with no output (exceeded 10m0s)
```

## Expected Behavior

Packages should be updated, no timeouts out

## Actual Behavior

Very often there is a timeout when upgrading packages, resulting in a failed `xenial-staging-with-rebase` step.

## Comments

--- a/.circleci/config.yml
+++ b/.circleci/config.yml
@@ -249,6 +249,7 @@ jobs:
       - run:
           name: Run Staging tests on GCE
           command: make ci-go
+          no_output_timeout: 20m
 
       - run:
           name: Ensure environment torn down

--- a/.circleci/config.yml
+++ b/.circleci/config.yml
@@ -276,6 +276,7 @@ jobs:
       - run:
           name: Run Staging tests on GCE (Xenial)
           command: make ci-go-xenial
+          no_output_timeout: 20m
 
       - run:
           name: Ensure environment torn down

